nntrainer/tensor/tensor_base.h

```diff
@@ -734,6 +721,32 @@ class TensorBase {
                            unsigned int &input_last_axis, unsigned int &M,
                            unsigned int &N, unsigned int &K, unsigned int &lda,
                            unsigned int &ldb, unsigned int &ldc) const;
+
+  /**
+   * @brief  Get the Data Type String object
+   * @return std::string of tensor data type
+   */
+  std::string getStringDataType() const {
```

Comment:
Instead of switch statement why don't we use function overriding as like other mathematical operations?

Reply:
Suggested code would be as follows.
```
class TensorBase {
  virtual std::string getStringDataType() const {
    return "Undefined type";
  }
}

class FloatTensor : public TensorBase {
  std::string getStringDataType() const override {
    return "FP32";
  }
}

class HalfTensor : public TensorBase {
  std::string getStringDataType() const override {
    return "FP16";
  }
}

...

```

Reply:
Good point! This could definitely make a better code structure.